kdc/krb5tgs.c

```diff
@@ -2182,8 +2182,8 @@ _kdc_tgs_rep(astgs_request_t r)
 				 r->armor_crypto,
 				 &req->req_body,
 				 r->error_code ? r->error_code : ret,
-				 r->client_princ,
-				 r->server_princ,
+				 r->client_princ ? r->client_princ :(r->ticket != NULL ? r->ticket->client : NULL),
+				 r->server_princ ? r->server_princ :(r->ticket != NULL ? r->ticket->server : NULL),
```

Comment:
This is ok.  It's definitely possible that `r->client_princ` and/or `r->server_princ` here are `NULL` because we're handling an error in some part of TGS request processing, and it's possible that we failed to validate the `PA-TGS`, say.  If `r->ticket` is not `NULL` it's because at least we managed to decrypt the ticket, so presumably we trust its contents, and that would be these principal names.  We're only going to use those principals in making a `KRB-ERROR`.

There's no privacy leak because `_kdc_fast_mk_error()` supports not exposing those names on the outer `KRB-ERROR`.